### PR TITLE
Add dlist instance

### PR DESCRIPTION
**Pull request by clintonmead**

### CODE REVIEW

File: autodocodec/package.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ library:
   - aeson
   - bytestring
   - containers
+  - dlist
```
Comment (NorfairKing) [inline autodocodec/package.yaml:23]: Do we already (transitively) depend on `dlist`?

Comment (clintonmead) [reply autodocodec/package.yaml:23]: I'm not sure. But I presume we need to explicitly depend on it to create the instances.

`dlist` is pretty lightweight, the only thing it depends on outside of what is distributed with GHC is `deepseq`. And `deepseq` itself depends only on `base`. Which is why I didn't create a new library/orphans.

Comment (NorfairKing) [reply autodocodec/package.yaml:23]: Ah yes `aeson` depends on `dlist so then this is fine.

**Comment by NorfairKing** [comment]
@clintonmead Looks good but we still need:

1.
```
nix flake lock --update-input validity
```
2. Upgrade stack.yaml to not depend on your fork but upstream validity
3. Minor version bump
4. Changelog entry

**Comment by clintonmead** [comment]
@NorfairKing  You'll need to make sure that has the right commit is merged into `validity`, it looks like the latest commit hasn't been merged into `validity` master: https://github.com/NorfairKing/validity/pull/120

Also I do have a changelog entry in Autodocodec, do you need it anywhere else (other than adjusting it after the minor version bump?

**Comment by NorfairKing** [comment]
@clintonmead should be done now.

**Comment by clintonmead** [comment]
@NorfairKing repushed with new validity commit hash

**Comment by NorfairKing** [comment]
Merged and released.

**Comment by NorfairKing** [comment]
Thank you ! :tada:

**Pull request merged**